nncf/quantization/algorithms/weight_compression/scale_estimation.py

```diff
@@ -0,0 +1,288 @@
+TTensor = TypeVar("TTensor")
+TWeightType = TypeVar("TWeightType")
+
+
+class ScaleEstimation(Algorithm):
+    """
+    Modified AWQ algorithm implementation.
+    """
+
+    def __init__(
+        self,
+        model: TModel,
+        name_to_node_mapping: Dict[str, Any],
+        all_weight_params: List[WeightCompressionParameters],
+        nodes_to_compress: List[NNCFNode],
+        activations: Optional[Dict[str, TTensor]] = None,
+        subset_size: int = 32,
+        initial_steps: int = 5,
+        scale_steps: int = 10,
+    ):
+        """
+        :param model: Model for applying algorithm.
+        :param name_to_node_mapping: Name to node mapping for updating node weights.
+        :param all_weight_params: List of all weight parameters.
+        :param nodes_to_compress: List of nodes for processing.
+        :param activations: The input activations of the layers considered for compression.
+        :param subset_size: The number of samples for scale estimation.
+        :param initial_steps: The number of the steps for absmax scale rectification.
+        :param scale_steps: The number of the steps for grid search scale rectification
+                            from 1.0 to 1.0 - 0.05 * scale_step.
+        """
+        super().__init__()
+        self.name_to_node_mapping = name_to_node_mapping
+        self._all_weight_params = all_weight_params
+        self._nodes_to_compress = nodes_to_compress
+        self._activations = activations
+        self._subset_size = subset_size
+        self._initial_steps = initial_steps
+        self._scale_steps = scale_steps
+
+        self._set_backend_entity(model)
+
+    @property
+    def available_backends(self) -> List[BackendType]:
+        return [BackendType.OPENVINO]
+
+    def _set_backend_entity(self, model: TModel) -> None:
+        """
+        Creates a helper class with a backed-specific logic of the algorithm.
+
+        :param model: Backend-specific input model.
+        :param all_weight_params: List of all weight parameters.
+        :param nodes_to_compress: List of nodes for processing.
+        :param activations: The input activations of the layers considered for compression.
+        """
+
+        model_backend = get_backend(model)
+        if model_backend == BackendType.OPENVINO:
+            from nncf.quantization.algorithms.weight_compression.openvino_backend import OVWeightCompressionAlgoBackend
+
+            self._backend_entity = OVWeightCompressionAlgoBackend(model, self.name_to_node_mapping)
+        else:
+            raise RuntimeError(
+                "Cannot return backend-specific AWQ entity because {} is not supported!".format(model_backend.value)
+            )
+
+    def apply(
+        self,
+        model: TModel,
+        graph: NNCFGraph,
+        statistic_points: Optional[StatisticPointsContainer] = None,
+        dataset: Optional[Dataset] = None,
+    ) -> TModel:
+        """
+        Applies the algorithm to the model.
```

Comment:
Could you add high-level comments, how is the algorithm estimates scales. Maybe some formulas or inline comments through the code are needed.

Reply:
Done